src/block/MossyCarpet.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\BlockEventHelper;
+use pocketmine\block\utils\StaticSupportTrait;
+use pocketmine\block\utils\WallConnectionType;
+use pocketmine\data\runtime\RuntimeDataDescriber;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\BlockTransaction;
+
+class MossyCarpet extends Flowable{
+	use StaticSupportTrait;
+
+	protected bool $top = false;
+
+	/** @var WallConnectionType[] */
+	protected array $sides = [];
+
+	protected function describeBlockOnlyState(RuntimeDataDescriber $w) : void{
+		$w->wallConnections($this->sides);
+		$w->bool($this->top);
+	}
+
+	public function getSideConnection(int $face) : ?WallConnectionType{
+		return $this->sides[$face] ?? null;
+	}
+
+	public function setSideConnection(int $face, ?WallConnectionType $value) : self{
+		if($value === null){
+			unset($this->sides[$face]);
+			return $this;
+		}
+		$this->sides[$face] = $value;
+		return $this;
+	}
+
+	public function isTop() : bool{ return $this->top; }
```

Comment:
This doesn't make sense to me. Since it actually decides whether or not the block is a carpet. Given the design of the states, it's also probably possible to stack multiple "top" blocks to get vines taller than 2 blocks from one carpet. Whether or not this _should_ be possible is another question.

We probably should split this into Pale Moss Carpet and Pale Moss (Carpet) Vines.